util/ranger/points.go

```diff
@@ -250,6 +250,28 @@ func (r *builder) buildFormBinOp(expr *expression.ScalarFunction) []point {
 		return nil
 	}
 
+	// if the colunm is unsigned integer and the value
+	// is less than zero, we only need compare with zero.
+	if mysql.HasUnsignedFlag(ft.Flag) {
+		if mysql.IsIntegerType(ft.Tp) && value.Kind() == types.KindInt64 {
+			zeroValue := new(types.Datum)
+			zeroValue.SetUint64(0)
+			cmp, err := value.CompareDatum(r.sc, zeroValue)
```

Comment:
Why not compare `value.GetInt64()` with 0 directly?

Reply:
ok